machine/v1beta1/types_machine.go

```diff
@@ -287,6 +315,7 @@ type LifecycleHook struct {
 }
 
 // MachineStatus defines the observed state of Machine
+// +openshift:validation:FeatureGateAwareXValidation:featureGate=MachineAPIMigration,rule="!has(oldSelf.synchronizedGeneration) || (has(self.synchronizedGeneration) && self.synchronizedGeneration >= oldSelf.synchronizedGeneration) || (oldSelf.authoritativeAPI == 'Migrating' && self.authoritativeAPI != 'Migrating')",message="syncronizedGeneration must not decrease unless authoritativeAPI is transitioning from Migrating to another value"
```

Comment:
I'll trust that there's a test for this.

Reply:
There are 8 for Machine and 8 for MachineSet specifically testing this rule